edgedb/retry.py

```diff
@@ -37,11 +39,17 @@ async def __aexit__(self, extype, ex, tb):
             # to propagate it to cancel the whole task.
             # NOTE: rollback error is always swallowed, should we use
             # on_log_message for it?
+        except ConnectionError:
+            reconnect = True
+
+        if reconnect:
+            await self._owner._reconnect()
```

Comment:
Hmm what if the `_reconnect()` fails again, shall we continue with the retry loop?

Reply:
We should retry according to the retry policy.  In fact, it is likely that a reconnect fails if there are network issues that caused the original network error in the first place.

Reply:
changed. I'm not 100% I got this part right. @fantix pleas have a look :mag: